lib/ansible/plugins/action/wait_for_connection.py

```diff
@@ -42,13 +47,27 @@ class ActionModule(ActionBase):
     DEFAULT_SLEEP = 1
     DEFAULT_TIMEOUT = 600
 
-    def do_until_success_or_timeout(self, what, timeout, connect_timeout, what_desc, sleep=1):
+    def do_until_success_or_timeout(self, what, timeout, ping_connect_timeout, what_desc, sleep=1):
         max_end_time = datetime.utcnow() + timedelta(seconds=timeout)
 
-        e = None
-        while datetime.utcnow() < max_end_time:
```

Comment:
I think you could simplify this. The `_total_seconds` shim doesn't seem necessary since those Python versions aren't supported anyway.

```suggestion
        while datetime.utcnow() < max_end_time:
            if (datetime.utcnow() + timedelta(seconds=connect_timeout)) > max_end_time:
                # connect_timeout is longer than the max time left, so use the remaining time instead
                connect_timeout = int((max_end_time - datetime.utcnow()).total_seconds())
            if connect_timeout < 1:
                # invalid timeout or only a fraction of a second is left
                break
```

Reply:
oh that's much simpler. I modified the suggestion slightly so it doesn't reassign to the connect_timeout argument and so it uses the same value for 'now' throughout an iteration of the while loop (good practice for guarding against timing edge cases).